src/lib/datepicker/datepicker.spec.ts

```diff
@@ -433,6 +433,58 @@ describe('MdDatepicker', () => {
         expect(testComponent.datepicker._minDate).toEqual(new Date(2010, JAN, 1));
         expect(testComponent.datepicker._maxDate).toEqual(new Date(2020, JAN, 1));
       });
+
+      it('should mark input invalid', async(() => {
```

Comment:
tried it but got errors when split, will debug it more tomorrow to find the cause and reply.